add-on/src/lib/redirect-handler/blockOrObserve.ts

```diff
@@ -119,6 +134,84 @@ function constructRegexFilter ({ originUrl, redirectUrl }: redirectHandlerInput)
   regexSubstitution: string
   regexFilter: string
 } {
+  let regexSubstitution = redirectUrl
+  let regexFilter = originUrl
+  const originURL = new URL(originUrl)
+  const redirectNS = computeNamespaceFromUrl(redirectUrl)
+  const originNS = computeNamespaceFromUrl(originUrl)
+  if (!DEFAULT_NAMESPACES.has(originNS) && DEFAULT_NAMESPACES.has(redirectNS)) {
+    // A redirect like https://github.com/ipfs/ipfs-companion/issues/1255
+    regexFilter = `^${escapeURLRegex(regexFilter)}`.replace(/https?/ig, 'https?')
+    const origRegexFilter = regexFilter
+
+    const [tld, root, ...subdomain] = originURL.hostname.split('.').reverse()
+    const staticUrl = [root, tld]
+    while (subdomain.length > 0) {
+      const subdomainPart = subdomain.shift()
```

Comment:
I believe `subdomain` is probably type `(string|undefined)[]` and needs a check to confirm it's not null, despite the fact that we confirm length in the while condition. We can explicitly set type of `subdomainPart` to string to avoid `as string` twice below.

Or, A potentially more type-safe approach could be to do something like

```
let subdomainPart = subdomain.shift()
while(subdomainPart != null) {
// ... code
 subdomainPart = subdomain.shift()
}